widgetssdk/src/main/java/com/glia/widgets/di/RepositoryFactory.java

```diff
@@ -99,7 +99,9 @@ public GliaQueueRepository getGliaQueueRepository() {
 
     public FileAttachmentRepository getGliaFileAttachmentRepository() {
         if (fileAttachmentRepository == null) {
-            fileAttachmentRepository = new FileAttachmentRepository(gliaCore);
+            fileAttachmentRepository = new FileAttachmentRepository(gliaCore,
+                    getGliaEngagementTypeRepository(),
+                    () -> new GliaCoreImpl().getSecureConversations());
```

Comment:
`new GliaCoreImpl().getSecureConversations()` can be changed to `gliaCore.getSecureConversations()`, and also I don't understand why was lambda used instead of passing direct instance.

Reply:
The lambda uses because we can't get secure messaging before the SDK is not initialized.

Reply:
What do you think? maybe it makes sense to move the initialization check from `getSecureConversations()` to methods inside `SecureConversations`, I myself removed a couple of that checks because we had that check when getting instances, but It seems this check is not mandatory at this step and making obstacles for integrators
cc: @andrews-moc @gugalo

Reply:
Honestly, I don't understand why and how can `getSecureConversations()` be called before SDK is initialized. Maybe I am missing anything, could you please explain?

P.S. Andrii has just merged the PR.

Reply:
That's because Dependencies in Widgets SDK initializing in the `onAppCreate()` method, before Glia.initialize() call, and we need to pass SecureConversations instance to Repository constructor

Reply:
Your suggestion makes sense to me then.

Reply:
Okay, I will rise this question in the channel once again.